sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -181,22 +195,34 @@
      * or has been cancelled. The completed operation returns a List of {@link RecognizedForm}.
      * @throws FormRecognizerException If recognize operation fails and the {@link AnalyzeOperationResult} returned with
      * an {@link OperationStatus#FAILED}.
-     * @throws NullPointerException If {@code recognizeOptions} is {@code null}.
+     * @throws NullPointerException If {@code form}, {@code modelId} is null.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PollerFlux<OperationResult, List<RecognizedForm>>
-        beginRecognizeCustomForms(Flux<ByteBuffer> form, long length, String modelId,
-        RecognizeOptions recognizeOptions) {
+    public PollerFlux<OperationResult, List<RecognizedForm>> beginRecognizeCustomForms(
+        Flux<ByteBuffer> form, long length, String modelId, RecognizeOptions recognizeOptions) {
         try {
+            Objects.requireNonNull(form, "'form' is required and cannot be null.");
+            Objects.requireNonNull(modelId, "'modelId' is required and cannot be null.");
+
             recognizeOptions = getRecognizeOptionsProperties(recognizeOptions);
```

Comment:
parameter `recognizeOptions` is re-assigned before being used. Is this intentional?